src/map.cpp

```diff
@@ -8424,3 +8424,13 @@ void map::clip_to_bounds( int &x, int &y, int &z ) const
         z = OVERMAP_HEIGHT;
     }
 }
+
+bool map::has_nearby_water( const tripoint &p, int radius )
```

Comment:
You probably don't need this at all, just use the crafting inventory and check for any amount of water.

```C++
    const inventory &crafting_inv = p->crafting_inventory();
    if( !crafting_inv.has_amount( "water", 100 ) || !crafting_inv.has_amount( "water_clean", 100 ) ) {
        add_msg("You need water");
    }

    ... // do the taks, add more checks ...

    std::vector<item_comp> comps;
    comps.push_back( item_comp( "water", 100 ) );
    comps.push_back( item_comp( "water_clean", 100 ) );
    p->consume_items( comps );
```

This will use any nearby water, be it natural source (river, pond) or a container.